org.eclipse.xtext.ui.testing/src/org/eclipse/xtext/ui/testing/AbstractContentAssistTest.java

```diff
@@ -104,4 +124,132 @@ protected void initializeTypeProvider(XtextResourceSet resourceSet) {
 		resourceSet.setClasspathURIContext(getJavaProject(resourceSet));
 	}
 
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension).
+	 * 		To be able to specify the workspace relative path,
+	 * 		use {@link IResourcesSetupUtil#createFile(String, String)} instead.
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, CharSequence content) {
+		return createDslFile(fileName, fileExtensionProvider.getPrimaryFileExtension(), content);
+	}
+
+	/**
+	 * Creates a dsl file.
+	 * 
+	 * @param fileName
+	 * 		The name of the file (without the file extension)
+	 * 
+	 * @param fileExtension
+	 * 		The extension of the file.
+	 * 
+	 * @param content
+	 * 		The content of the file.
+	 * 
+	 * @since 2.35
+	 */
+	protected IFile createDslFile(String fileName, String fileExtension, CharSequence content) {
+		try {
+			return IResourcesSetupUtil.createFile(javaProject.getElementName(), fileName, fileExtension, content.toString());
+		} catch (InvocationTargetException | CoreException | InterruptedException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * Verifies whether the content assistant provides the expected proposals on the given cursor position.
+	 * 
+	 * @param text
+	 * 		The editor's input text. The text must contain the {@link #c} special symbol indicating the current cursor position.
+	 * 
+	 * @param expectedProposals
+	 * 		The proposals (separated by new lines) that are expected to be offered whenever the content assistant is triggered on the given cursor position.
+	 * 
+	 * @since 2.35
+	 */
+	protected void assertContentAssistant(CharSequence text, CharSequence expectedProposals) {
+		assertContentAssistant(text, expectedProposals.toString().split(System.lineSeparator()), null, null);
```

Comment:
Maybe I opened a Pandora's box... O:)
I'd like to avoid introducing further problems due to d**n line endings in Windows.
Your implementation works when using Xtend template strings but would fail when using Java text blocks or Xtend multi-line strings that are now (https://github.com/eclipse/xtext/pull/3004) normalized w.r.t. line endings (only Unix EOL is generated).

I'd like this solution to be guarded against different EOLs...

A possible solution would be to first normalize the `expectedProposals` with only Unix EOL (we have a utility method `Strings.toUnixLineSeparator`) and then split it only according "\n".

I haven't tried that, but it should work.

Reply:
@LorenzoBettini : I have just modified the PR as requested:

```
	protected void assertContentAssistant(CharSequence text, CharSequence expectedProposals) {
		String[] expectedProposalsArray = Strings.toUnixLineSeparator(expectedProposals.toString()).split("\n");
		assertContentAssistant(text, expectedProposalsArray, null, null);
	}

	protected void assertContentAssistant(CharSequence text, CharSequence expectedProposals, String proposalToApply, String expectedContent) {
		String[] expectedProposalsArray = Strings.toUnixLineSeparator(expectedProposals.toString()).split("\n");
		assertContentAssistant(text, expectedProposalsArray, proposalToApply, expectedContent);
	}
```

Reply:
@miklossy Thanks! I feel safer now ;)

Just in case, can't the first method be implemented in terms of the second one, by simply passing `text, expectedProposals, null, null`?